lisa/sut_orchestrator/qemu/platform.py

```diff
@@ -360,16 +404,20 @@ def _delete_nodes(
             except Exception as ex:
                 log.warning(f"console log delete failed. {ex}")
 
-            # Delete the files created for the VM.
-            try:
-                os.remove(node_context.os_disk_file_path)
-            except Exception as ex:
-                log.warning(f"OS disk delete failed. {ex}")
+            if self.qemu_platform_runbook.is_host_remote():
+                vm_dir = os.path.dirname(node_context.os_disk_file_path)
```

Comment:
We should avoid having different logic for local and remote hosts. Either both local and remote hosts should create sub-directories for each VM or neither should.